### PR TITLE
[REVIEW] Add nightly version for ucx-py in ci script

### PR DESCRIPTION
**Pull request by galipremsagar**
`ucx-py` recently reverted back to old versioning format from calVer. This PR hard-codes the nightly version.

### CODE REVIEW

**Review by dillon-cullinan — APPROVED** [review]

**Review by jakirkham — APPROVED** [review]

File: ci/gpu/build.sh [hunk 1]
```diff
@@ -80,7 +80,7 @@ gpuci_conda_retry install -y \
                   "rapids-notebook-env=$MINOR_VERSION.*" \
                   "dask-cuda=${MINOR_VERSION}" \
                   "rmm=$MINOR_VERSION.*" \
-                  "ucx-py=${MINOR_VERSION}"
+                  "ucx-py=0.21.0a*"
```
Comment (ajschmidt8) [inline ci/gpu/build.sh:83]: @jakirkham @galipremsagar, do we need the `.0a` suffix here? or can it just be `0.21.*`?

Comment (jakirkham) [reply ci/gpu/build.sh:83]: I think `0.21.*` should be ok

Comment (galipremsagar) [reply ci/gpu/build.sh]: Done 👍

Comment (jakirkham) [reply ci/gpu/build.sh]: Thanks Prem 🙂

**Review by jakirkham — APPROVED** [review]

**Review by jakirkham — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]

**Comment by harrism** [comment]
@gpucibot merge

**Pull request merged**